docs/proposals/05-etcd-operator-tasks.md

```diff
@@ -0,0 +1,424 @@
+
+  // OwnerEtcdReference refers to the name and namespace of the corresponding 
+  // Etcd owner for which the task has been invoked.
+  OwnerEtcdRefrence types.NamespacedName `json:"ownerEtcdRefrence"`
+}
+```
+
+#### Status
+
+The authors propose that the following fields should be specified in the `Status` (current state) of the `EtcdOperatorTask` custom resource as the custom resource's `Status` will be used to monitor the progress of the task.
+
+* To capture the current state of a task, a field `.status.currentState` is defined in status.
+* If an operation involves many stages, `.status.lastOperation` will be useful in capturing the status of any intermediate stage.
+
+```go
+// EtcdOperatorTaskStatus is the status for a EtcdOperatorTask resource.
+type EtcdOperatorTaskStatus struct {
+  // ObservedGeneration is the most recent generation observed for the resource.
+  ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+  // State is the last known state of the task.
+  State TaskState `json:"state"`
+  // Time at which the task has moved from "pending" state to any other state.
+  InitiatedAt metav1.Time `json:"initiatedAt"`
+  // LastError represents the errors when processing the task.
+  LastErrors []LastError `json:"lastErrors,omitempty"`
+  // Captures the last operation.
+  // +optional
+  LastOperation *LastOperation `json:"lastOperation,omitempty"`
+}
+
+type LastOperation struct {
+  // Name of the LastOperation.
+  Name opsName `json:"name"`
+  // Status of the last operation, one of pending, progress, completed, failed.
+  State OperationState `json:"state"`
+  // LastTransitionTime is the time at which the operation state last transitioned from one state to another.
+  LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+  // A human readable message indicating details about the last operation.
+  Reason string `json:"reason"`
+}
+
+// LastError stores details of the most recent error encountered for the task.
+type LastError struct {
+  // Code is an error code that uniquely identifies an error.
+  Code ErrorCode `json:"code"`
+  // Description is a human-readable message indicating details of the error.
+  Description string `json:"description"`
+  // ObservedAt is the time at which the error was observed.
+  ObservedAt metav1.Time `json:"observedAt"`
+}
+
+// TaskState represents the state of the task.
+type TaskState string
+
+const (
+  Failed TaskState = "Failed"
+  Pending TaskState = "Pending"
+  Rejected TaskState = "Rejected"
+  Succeeded TaskState = "Succeeded"
+  InProgress TaskState = "InProgress"
+)
+
+// OperationState represents the state of last operation.
+type OperationState string
+
+const (
+  Failed OperationState = "OperationStateFailed"
+  Pending OperationState = "OperationStatePending"
+  Completed OperationState = "OperationStateCompleted"
+  InProgress OperationState = "OperationStateInProgress"
+)
+```
+
+### Custom Resource API object
```

Comment:
The title should be `Custom Resource Definition`

Reply:
It's CR object not the CRD ... right ?